Tools/WebKitTestRunner/TestController.cpp

```diff
@@ -360,8 +361,27 @@ void TestController::setIsMediaKeySystemPermissionGranted(bool granted)
     m_isMediaKeySystemPermissionGranted = granted;
 }
 
-static void queryPermission(WKStringRef, WKSecurityOriginRef, WKQueryPermissionResultCallbackRef callback)
+static void queryPermission(WKStringRef string, WKSecurityOriginRef securityOrigin, WKQueryPermissionResultCallbackRef callback)
 {
+    TestController::singleton().handleQueryPermission(string, securityOrigin, callback);
+}
+
+void TestController::handleQueryPermission(WKStringRef string, WKSecurityOriginRef securityOrigin, WKQueryPermissionResultCallbackRef callback)
+{
+    if (toWTFString(string) == "notifications"_s) {
+        auto securityOriginString = adoptWK(WKSecurityOriginCopyToString(securityOrigin));
+        auto dictionary = adoptWK(m_webNotificationProvider.notificationPermissions());
+
+        if (auto value = WKDictionaryGetItemForKey(dictionary.get(), securityOriginString.get())) {
+            bool permissionWasGranted = WKBooleanGetValue(static_cast<WKBooleanRef>(value));
+            if (permissionWasGranted == 1)
```

Comment:
Like I said in my earlier review, no `== 1`

Reply:
Fixed!